### DISCUSSION TITLE
Difference between GCN and CNN

### DISCUSSION THREAD

**Discussion by meyerTUM** [pyg-team/pytorch_geometric#2179] [Q&A] [answered]
If a regular 2D grid with c channels is converted to a graph and used as input for gcn what is the difference to a cnn?

Do I understand it correctly, that it the gcn weights would correspond to 3x3 kernels in the cnn where the corner weights are 0 and all other 5 weights are identical?

**Comment by rusty1s** [comment, marked as answer]
By default, the GCN is isotropic, which means that it will not learn distinct weights depending on the relative position of nodes. You may see it as `3x3` kernel with identical values. 

This observation was one of the motivations for our [`SplineCNN` paper](https://arxiv.org/abs/1711.08920), which conditions transformations based on the relative positioning of neighboring nodes.

**Reply by meyerTUM** [reply]
Thank you for confirming that. Really great work (paper + pytorch geometric)!
